File: kawazoe/q2029/tyama_codeiq2029.java
import java.util.*;
class Main{
	public static void main(String[]args){
		Scanner cin=new Scanner(System.in);
		for(;cin.hasNext();){
			long n=cin.nextLong();
			if(n==0)break;
			long r=0;
			int i=2;
			for(;;i++){
				long x=n-(long)i*(i-1)/2;
				if(x<i)break;
				if(x%i==0)r+=x/i;
			}
			System.out.println(r);
		}
	}
}

File: kawazoe/q2029/tyama_codeiq2029.scala
//usr/bin/env scala $0 $@;exit
object Main{
	def calc(r:Long,i:Long,n:Long):Long = {
		val x = n-i*(i-1)/2
		if(x>=i){
			if(x%i==0){
				calc(r+x/i,i+1,n)
			}else{
				calc(r,i+1,n)
			}
		}else{
			r
		}
	}

	def main(args: Array[String]) = {
		val cin=new java.util.Scanner(System.in)
		while(cin.hasNext()){
			var n=cin.nextLong()
			if(n>0)println(calc(0,2,n))
		}
	}
}